### PR TITLE
feat: integrate backend for the login flow

### PR DESCRIPTION
**Pull request by Mokto**

### CODE REVIEW

File: src/api/base.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import Axios from 'axios';
 
-const baseUrl = 'http://192.168.0.122:3100';
+// const baseUrl = 'http://192.168.0.122:3100';
```
Comment (mfateminia) [inline src/api/base.ts]: comment?

File: ios/komak/Info.plist [hunk 2]
```diff
@@ -27,12 +27,12 @@
 			<string>Editor</string>
 			<key>CFBundleURLSchemes</key>
 			<array>
-				<string>com.googleusercontent.apps.153461499435-0kruiebmi9ikjpg4fv531eli76oe1kgr</string>
+				<string>com.googleusercontent.apps.50726922019-rm0ehq9702sgre48e6c6anou4sk6i4jg</string>
```
Comment (mfateminia) [inline ios/komak/Info.plist]: is this on purpose?

File: src/models/profile.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+interface Location {
+  type: 'Point';
+  coordinates?: [number, number];
```
Comment (mfateminia) [inline src/models/profile.ts]: not optional right?

File: src/stores/social-login-store.ts [hunk 4]
```diff
@@ -19,58 +19,66 @@ export class SocialLoginStore {
...
-      //   offlineAccess: false, // if you want to access Google API on behalf of the user FROM YOUR SERVER
-      //   hostedDomain: '', // specifies a hosted domain restriction
-      //   loginHint: '', // [iOS] The user's ID, or email address, to be prefilled in the authentication UI if possible. [See docs here](https://developers.google.com/identity/sign-in/ios/api/interface_g_i_d_sign_in.html#a0a68c7504c31ab0b728432565f6e33fd)
-      forceCodeForRefreshToken: true, // [Android] related to `serverAuthCode`, read the docs link below *.
-      //   accountName: '', // [Android] specifies an account name on the device that should be used
-      iosClientId:
-        '153461499435-0kruiebmi9ikjpg4fv531eli76oe1kgr.apps.googleusercontent.com', // [iOS] optional, if you want to specify the client ID of type iOS (otherwise, it is taken from GoogleService-Info.plist)
+      forceCodeForRefreshToken: true,
+      // iosClientId:
+      //   '153461499435-0kruiebmi9ikjpg4fv531eli76oe1kgr.apps.googleusercontent.com',
```
Comment (mfateminia) [inline src/stores/social-login-store.ts]: comment?

File: src/stores/social-login-store.ts [hunk 5]
```diff
@@ -19,58 +19,66 @@ export class SocialLoginStore {
...
-  public async googleLogin() {
+  public async googleLogin(): Promise<LoginResult> {
     try {
-      console.log('user info');
       //   await GoogleSignin.hasPlayServices();
       const userInfo = await GoogleSignin.signIn();
+      const firstName = userInfo.user.givenName;
+      const lastName = userInfo.user.familyName;
+      // const email = userInfo.email;
       console.log(userInfo);
```
Comment (mfateminia) [inline src/stores/social-login-store.ts]: comment and log

File: src/stores/social-login-store.ts [hunk 6]
```diff
@@ -19,58 +19,66 @@ export class SocialLoginStore {
...
     } catch (error) {
       if (error.code === statusCodes.SIGN_IN_CANCELLED) {
-        // user cancelled the login flow
+        return null;
       } else if (error.code === statusCodes.IN_PROGRESS) {
         // operation (e.g. sign in) is in progress already
       } else if (error.code === statusCodes.PLAY_SERVICES_NOT_AVAILABLE) {
         // play services not available or outdated
       } else {
         // some other error happened
```
Comment (mfateminia) [inline src/stores/social-login-store.ts]: ?

File: src/api/base.ts (hunk 1 above)
Comment (Mokto) [reply src/api/base.ts]: On purpose. We might have to switch between both for dev purposes.

File: ios/komak/Info.plist (hunk 2 above)
Comment (Mokto) [reply ios/komak/Info.plist]: Yes

File: src/stores/social-login-store.ts (hunk 6 above)
Comment (Mokto) [reply src/stores/social-login-store.ts]: I need to take care of those cases... later

**Pull request merged**